packages/blockchain-common/test/environments/AnsibleEnvironment.test.ts

```diff
@@ -12,127 +12,104 @@
-            mockFabricWalletGenerator.getWallet.should.have.been.calledOnceWithExactly('myWallet');
-            mockFabricWallet.importIdentity.should.have.been.calledOnceWithExactly(sinon.match.string, sinon.match.string, 'admin', 'Org1MSP');
-        });
+    describe('#getWalletsAndIdentities', () => {
+
+        it('should create all wallets and import all identities and return registry entries', async () => {
+            const importIdentityStub: sinon.SinonStub = sinon.stub().resolves();
+            const mockFabricWallet: any = {
+                importIdentity: importIdentityStub
+            };
+
+            const mockFabricWalletGenerator: any = {
+                getWallet: sinon.stub().resolves(mockFabricWallet)
+            };
 
-        it('should import identities if wallet already exists', async () => {
-            const walletEntry: FabricWalletRegistryEntry = new FabricWalletRegistryEntry();
-            walletEntry.name = 'myWallet';
-            await FabricWalletRegistry.instance().add(walletEntry);
+            FabricWalletGeneratorFactory.setFabricWalletGenerator(mockFabricWalletGenerator);
 
-            const mockFabricWalletGenerator: sinon.SinonStubbedInstance<IFabricWalletGenerator> = sandbox.createStubInstance(FabricWalletGenerator);
-            sandbox.stub(FabricWalletGeneratorFactory, 'createFabricWalletGenerator').returns(mockFabricWalletGenerator);
-            const mockFabricWallet: sinon.SinonStubbedInstance<IFabricWallet> = sandbox.createStubInstance(FabricWallet);
-            mockFabricWalletGenerator.getWallet.returns(mockFabricWallet);
-            await environment.importWalletsAndIdentities();
-            mockFabricWalletGenerator.getWallet.should.have.been.calledOnceWithExactly('myWallet');
+            const entries: FabricWalletRegistryEntry[] = await environment.getWalletsAndIdentities();
+
+            entries.length.should.equal(1);
+            entries[0].name.should.equal('myWallet');
+            entries[0].walletPath.should.equal(path.join(environmentPath, FileConfigurations.FABRIC_WALLETS, 'myWallet'));
+            entries[0].fromEnvironment.should.equal('nonManagedAnsible');
+            mockFabricWalletGenerator.getWallet.should.have.been.calledOnceWithExactly(sinon.match.instanceOf(FabricWalletRegistryEntry));
             mockFabricWallet.importIdentity.should.have.been.calledOnceWithExactly(sinon.match.string, sinon.match.string, 'admin', 'Org1MSP');
         });
-
     });
 
-    describe('#importGateways', () => {
-        it('should create all gateways', async () => {
-            const extDir: string = vscode.workspace.getConfiguration().get(SettingConfigurations.EXTENSION_DIRECTORY);
-            const homeExtDir: string = FileSystemUtil.getDirPath(extDir);
+    describe('#getGateways', () => {
+        it('should get all gateways', async () => {
 
-            await environment.importGateways();
+            await environment.getGateways();
```

Comment:
You've called this but not used it (as you call it again on line 77)

Reply:
done